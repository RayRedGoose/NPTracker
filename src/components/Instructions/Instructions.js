import './Instructions.scss'
import React, { Component } from 'react'

class Instructions extends Component {
  constructor() {
    super()
    this.state = {
      text: 'Click on \'Dashboard\' tab to see your summary',
      count: 2
    }
  }

  componentDidMount() {
    const texts = [
      'Click on \'Dashboard\' tab to see your summary',
      'Click on \'Parks\' tab to see all national parks',
      'Click on \'Wish List\' tab to see your plannings',
      'Click on \'Visited\' tab to see your visited places'
    ]

    setInterval(() => {
      let { count } = this.state
      return (count !== 4)
        ? this.setState({text: texts[count], count: count+1})
        : this.setState({text: texts[0], count: 1})
    }, 6000)
  }

  render() {
    return (
      <section className="instructions">
        <h2>{this.state.text}</h2>
      </section>
    )
  }
}

export default Instructions
