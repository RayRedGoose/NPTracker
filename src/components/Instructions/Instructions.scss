@import "../../variables.scss";

.instructions {
  h2 {
    color: $accent-color-primary-light;
    font-family: Salsa;
    font-size: 36px;
    font-weight: 400;
    letter-spacing: 0.1em;
    margin: auto;
    text-align: center;
    animation: shows 6s linear infinite;
    position: absolute;
    top: 45%;
    width: 85%;
  }
}

@keyframes shows {
  0% {
    opacity: 0;
  }
  40% {
    opacity: 1;
  }
  80% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
