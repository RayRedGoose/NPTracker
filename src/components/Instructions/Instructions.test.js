import React from 'react'
import { shallow } from 'enzyme'
import Instructions from './Instructions'

describe('Instructions', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const block = shallow(<Instructions />)

    expect(block).toMatchSnapshot()
  })
})
