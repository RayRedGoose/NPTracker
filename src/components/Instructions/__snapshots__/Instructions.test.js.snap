// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions should match the snapshot with all data passed in correctly 1`] = `
<section
  className="instructions"
>
  <h2>
    Click on 'Dashboard' tab to see your summary
  </h2>
</section>
`;
